tasks/test_add_picklist_values.py

```diff
@@ -0,0 +1,412 @@
+                    "valueName": "Test"
+                }]
+            },
+                {
+                "picklist": "hed__Attribute_Type__c",
+                "values": [{
+                    "default": False,
+                    "valueName": "Existing Value"
+                }]
+            }]
+        },
+        "Name": "Student Characteristic"
+    }],
+    "size": 2,
+}
+
+api_version = "47.0"
+
+
+class TestAddPicklistValues(unittest.TestCase):
+
+    def _get_base_tooling_url(self, instance_url):
+        return "{}/services/data/v{}/tooling/".format(
+            instance_url,
+            api_version
+        )
+
+    def _get_base_deployment_url(self, instance_url, org_id):
+        return "{}/services/Soap/m/{}/{}".format(
+            instance_url,
+            api_version,
+            org_id
+        )
+
+    @responses.activate
+    def test_basic(self):
+        add_picklist_values_task = create_task(
+            add_picklist_values.AddPicklistValues, task_options)
+
+        base_tooling_url = self._get_base_tooling_url(
+            add_picklist_values_task.org_config.instance_url)
+
+        base_deployment_url = self._get_base_deployment_url(
+            add_picklist_values_task.org_config.instance_url,
+            add_picklist_values_task.org_config.org_id
+        )
+
+        customobject_query_url = (
+            base_tooling_url
+            + "query/?q=SELECT+DeveloperName%2C+DurableId+FROM+EntityDefinition+"
+            "WHERE+NamespacePrefix+%3D+%27hed%27+AND+DeveloperName+%3D+%27Attribute%27"
+        )
+
+        customfield_query_url = (
+            base_tooling_url
+            + "query/?q=SELECT+Id%2C+DeveloperName%2C+Metadata+FROM+CustomField+"
+            + "WHERE+NamespacePrefix+%3D+%27hed%27+AND+DeveloperName+%3D+%27Attribute_Type%27+AND+TableEnumOrId+%3D+%27123%27"
+        )
+
+        responses.add(
+            method=responses.GET,
+            url=customobject_query_url,
+            match_querystring=True,
+            json=expected_customobject_query_response
+        )
+
+        responses.add(
+            method=responses.GET,
+            url=customfield_query_url,
+            match_querystring=True,
+            json=expected_customfield_query_response
+        )
+
+        responses.add(
+            method=responses.POST,
+            url=base_deployment_url,
+            body=deploy_result.format(
+                status="Succeeded", extra="<id>123</id>"),
+            status=200,
+            content_type="text/xml; charset=utf-8",
+        )
+
+        add_picklist_values_task()
+
+        self.assertEqual(4, len(responses.calls))
+        self.assertEqual(customobject_query_url,
+                         responses.calls[0].request.url)
+        self.assertEqual(customfield_query_url, responses.calls[1].request.url)
+        self.assertEqual(base_deployment_url, responses.calls[2].request.url)
+        self.assertEqual(base_deployment_url, responses.calls[3].request.url)
```

Comment:
@davisagli The tests, even as simplistic as they are, have actually already been very valuable as I've made some minor updates to the logic over the past few days. In order to keep this moving, I would perhaps prefer to create a tech debt item to improve on the unit tests in the future if you think that's acceptable (and this would give other members of the EDA team a chance to work on CCI if they'd like)

Reply:
@davisagli One question I did have in regards to this, though: it's not clear to me why there are 4 calls here when my 3rd response (the deployment) returns the "Succeeded" status. Do you know why this is? It seems as though there are always 2 calls for a deployment and I haven't figured out why.

Reply:
@robertbwatson  In a situation like that, I might TEMPORARILY (!) add a couple of lines like:

print(responses.calls)
assert 0, responses.calls

And see if the output is clarifying. Have you tried something like that? Python tends to output a lot more information about objects than Apex or Javascript so often just "print" ing them will answer questions. (not always, of course...some objects have too much internal state to be easily print-able)

Reply:
@robertbwatson yeah that's fine. I'm guessing this is at a point where actually trying to use it for real would be as useful for testing as writing more tests. Also if we end up turning this into a task in CumulusCI, we can revisit the tests then.

Metadata API deployments are an asynchronous process. There are always at least 2 requests: one call to deploy which initiates it, and then at least one call to checkDeployStatus.